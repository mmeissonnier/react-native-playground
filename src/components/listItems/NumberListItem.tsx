import React, { useState, StatelessComponent } from 'react';
import { View, Text } from 'react-native';
import { Slider } from 'react-native-elements';
import { ListItemProps } from '../../types';

const NumberListItem: StatelessComponent<ListItemProps<number>> = ({
  id,
  label,
  value = 0,
  onChange,
}) => {
  const [state, setState] = useState({ value });

  return (
    <View
      style={{
        display: 'flex',
        flexDirection: 'row',
        width: '100%',
        alignItems: 'center',
      }}
    >
      <Text style={{ fontSize: 20, fontWeight: '200', color: 'white' }}>
        {label}
      </Text>
      <Slider
        style={{ width: 200 }}
        minimumValue={0}
        maximumValue={50}
        step={10}
        onValueChange={newValue => {
          setState({ value: newValue });
          if (onChange) {
            onChange(id, newValue);
          }
        }}
        value={state.value}
      />
    </View>
  );
};

export default React.memo(NumberListItem);
