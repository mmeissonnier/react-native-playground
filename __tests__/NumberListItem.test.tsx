import React from 'react';
import { shallow } from 'enzyme';
import NumberListItem from '../src/components/listItems/NumberListItem';

test('It should render correctly', () => {
  expect(shallow(<NumberListItem id="1" label="Foo" />)).toMatchSnapshot();
});
