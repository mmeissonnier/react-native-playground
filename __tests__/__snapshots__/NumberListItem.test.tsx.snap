// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 20,
        "fontWeight": "200",
      }
    }
  >
    Foo
  </Text>
  <ForwardRef(Themed.Slider)
    maximumValue={50}
    minimumValue={0}
    onValueChange={[Function]}
    step={10}
    style={
      Object {
        "width": 200,
      }
    }
    value={0}
  />
</View>
`;
